server/chat.ts

```diff
@@ -1107,6 +1107,29 @@ export class CommandContext extends MessageContext {
 
 		return message;
 	}
+	canPMHTML(targetUser: User | null) {
+		if (!targetUser || !targetUser.connected) {
+			this.errorReply(`User ${this.targetUsername} is not currently online.`);
+			return false;
+		}
+		if (!(this.room && (targetUser.id in this.room.users)) && !this.user.can('addhtml')) {
+			this.errorReply("You do not have permission to use this command to users who are not in this room.");
```

Comment:
The problem with this is: what if a user tries to join `view-bot-scrappie-test` without sharing any rooms with Scrappie?

Reply:
i mean, the obvious solution is just to stop you from using that unless the bot can PM you back.

alternately, i could see some kind of "which connection last requested which page" property being nice to have (for this and for #7051

Reply:
Yeah, I think that might be the best way to d oit.

Reply:
This part doesn't seem fixed?

Reply:
look at the sendhtmlpage function, this only gets called if there's no relevant connection to send the page to.

Reply:
Oh!